pypit/arsciexp.py

```diff
@@ -585,6 +551,7 @@ def MasterPinhole(self, fitsdict, det):
             debugger.set_trace()  # NEED TO LOAD EXTRAS AS ABOVE
         # Set and then delete the Master Trace frame
         self.SetMasterFrame(mspinhole, "pinhole", det)
+        armasters.save_masters(self, det, mftype='pinhole')
```

Comment:
Don't we want to put this line somewhere inside the exception (that starts on line 523)? Otherwise, we would be saving a master frame that either already exists (i.e. on line 522), or that we've loaded from a set filename (i.e. on line 550).